pkg/artifactory/datasource_artifactory_file.go

```diff
@@ -0,0 +1,202 @@
+package artifactory
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/atlassian/go-artifactory/v2/artifactory"
+	"github.com/atlassian/go-artifactory/v2/artifactory/v1"
+	"github.com/hashicorp/terraform/helper/schema"
+	"io"
+	"os"
+)
+
+func datasourceArtifactoryFile() *schema.Resource {
+	return &schema.Resource{
+		Create: nil,
+		Read:   resourceArtifactRead,
+		Update: nil,
+		Delete: nil,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modified_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_updated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"download_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mimetype": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"md5": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sha1": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sha256": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"output_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"force_overwrite": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceArtifactRead(d *schema.ResourceData, m interface{}) error {
```

Comment:
Looking at other providers the naming convention seems to be `dataSource` instead of `resource`. So this would become `dataSourceArtifactRead`

Reply:
Fixed with 29c8eeb